importers/link_16_importer.py

```diff
@@ -0,0 +1,199 @@
+        sensor = platform.get_sensor(
+            data_store=data_store,
+            sensor_name="Link-16",
+            sensor_type=sensor_type,
+            privacy=privacy,
+            change_id=change_id,
+        )
+        # -- END TODO
+        state = datafile.create_state(data_store, platform, sensor, line_time, self.short_name)
+
+        location = Location(errors=self.errors, error_type=self.error_type)
+        lat_success = location.set_latitude_decimal_degrees(lat_degrees_token.text)
+        lon_success = location.set_longitude_decimal_degrees(lon_degrees_token.text)
+        if lat_success and lon_success:
+            state.location = location
+            combine_tokens(lon_degrees_token, lat_degrees_token).record(
+                self.name, "location", state.location, "decimal degrees"
+            )
+
+        elevation_valid, elevation = convert_distance(
+            altitude_token.text, unit_registry.foot, line_number, self.errors, self.error_type
+        )
+
+        if elevation_valid:
+            state.elevation = elevation
+            altitude_token.record(self.name, "altitude", state.elevation)
+
+        heading_valid, heading = convert_absolute_angle(
+            heading_token.text, line_number, self.errors, self.error_type
+        )
+        if heading_valid:
+            state.heading = heading
+            heading_token.record(self.name, "heading", heading)
+
+        speed_valid, speed = convert_speed(
+            speed_token.text,
+            unit_registry.foot_per_second,
+            line_number,
+            self.errors,
+            self.error_type,
+        )
+        if speed_valid:
+            state.speed = speed
+            speed_token.record(self.name, "speed", speed)
+
+    @staticmethod
+    def extract_timestamp(filename):
+        """Extracts a Link-16 timestamp from an unmodified filename
+        :param filename: The Link-16 filename that contains a timestamp
+        :type filename: String
+        :return: The timestamp extracted from the filename
+        :rtype: String
+        """
+        # TODO - confirm that there is always a double extension
+        base_filename, ext = os.path.splitext(os.path.basename(filename))
+        base_filename, ext = os.path.splitext(base_filename)
+        # TODO - confirm the timestamp always just before the file extension
+        extracted_datetime = base_filename[-19:]
+        return extracted_datetime
+
+    @staticmethod
+    def timestamp_to_datetime(timestamp_string):
+        """Converts a Link-16 formatted timestamp into a Python datetime
+        :param timestamp_string: The Link16 file timestamp
+        :type timestamp_string: String
+        :return: a datetime if conversion successful or False if unsuccessful
+        :rtype: datetime | bool
+        """
+        timestamp_format = "%d-%m-%YT%H-%M-%S"
+        try:
+            res = datetime.strptime(timestamp_string, timestamp_format)
```

Comment:
while I have DST on my mind, is it worth explicitly specifying that this will be a GMT time?

Reply:
I've noted this in the doc-comment. Couldn't see an obvious way to indicate this with the strptime function, but happy to add if I've missed a parameter

Reply:
Aah, I know.  We can check this through inspection, since we're currently in DST.  If you print a `timestamp_string` and the resulting object to the command line - you can check they are the same value.  If they are - then we can be confident it is forcing it to UTC.